mcx/jax.py

```diff
@@ -71,3 +74,54 @@ def unravel_list(arr):
     )
 
     return flat, unravel_list
+
+
+def progress_bar_factory(num_samples):
+    """Factory that builds a progress bar decorator along
+    with the `set_tqdm_description` and `close_tqdm` functions
+    """
+
+    tqdm_pbar = tqdm(range(num_samples))
+
+    def set_tqdm_description(message):
+        tqdm_pbar.set_description(
+            message,
+            refresh=False,
+        )
+
+    def close_tqdm():
+        tqdm_pbar.close()
+
+    def _update_tqdm(arg, transform):
+        tqdm_pbar.update(arg)
+
+    @jit
+    def _progress_bar(arg, result):
```

Comment:
`_update_progress_bar`? Maybe we'll go over naming once it's ready to merge. I'm very particular about names because they tend to stay forever :fearful: